rskj-core/src/main/java/org/ethereum/config/SystemProperties.java

```diff
@@ -164,22 +123,58 @@ public SystemProperties(Config apiConfig) {
+            this.projectVersion = getProjectVersion(props);
+            this.projectVersionModifier = getProjectVersionModifier(props);
 
         } catch (Exception e) {
             logger.error("Can't read config.", e);
             throw new RuntimeException(e);
         }
     }
 
+    private static String getProjectVersion(Properties props) {
+        String versionNumber = props.getProperty("versionNumber");
+
+        if (versionNumber == null) {
+            return "-.-.-";
+        }
+
+        return versionNumber.replaceAll("'", "");
+    }
+
+    private static String getProjectVersionModifier(Properties props) {
+        return props.getProperty("modifier").replaceAll("\"", "");
+    }
+
+    private static Config getConfigFromFiles() {
+        Config javaSystemProperties = ConfigFactory.load("no-such-resource-only-system-props");
+        Config referenceConfig = ConfigFactory.parseResources("rskj.conf");
+        logger.info("Config ( {} ): default properties from resource 'rskj.conf'", referenceConfig.entrySet().isEmpty() ? NO : YES);
+        String res = System.getProperty("rskj.conf.res");
+        Config cmdLineConfigRes = res != null ? ConfigFactory.parseResources(res) : ConfigFactory.empty();
+        logger.info("Config ( {} ): user properties from -Drskj.conf.res resource '{}'", cmdLineConfigRes.entrySet().isEmpty() ? NO : YES, res);
+        Config userConfig = ConfigFactory.parseResources("user.conf");
+        logger.info("Config ( {} ): user properties from resource 'user.conf'", userConfig.entrySet().isEmpty() ? NO : YES);
+        File userDirFile = new File(System.getProperty("user.dir"), "/config/rskj.conf");
+        Config userDirConfig = ConfigFactory.parseFile(userDirFile);
+        logger.info("Config ( {} ): user properties from file '{}'", userDirConfig.entrySet().isEmpty() ? NO : YES, userDirFile);
+        Config testConfig = ConfigFactory.parseResources("test-rskj.conf");
+        logger.info("Config ( {} ): test properties from resource 'test-rskj.conf'", testConfig.entrySet().isEmpty() ? NO : YES);
+        Config testUserConfig = ConfigFactory.parseResources("test-user.conf");
+        logger.info("Config ( {} ): test properties from resource 'test-user.conf'", testUserConfig.entrySet().isEmpty() ? NO : YES);
+        String file = System.getProperty("rsk.conf.file");
+        Config cmdLineConfigFile = file != null ? ConfigFactory.parseFile(new File(file)) : ConfigFactory.empty();
+        logger.info("Config ( {} ): user properties from -Drsk.conf.file file '{}'", cmdLineConfigFile.entrySet().isEmpty() ? NO : YES, file);
+        return javaSystemProperties
+                .withFallback(cmdLineConfigFile)
+                .withFallback(testUserConfig)
+                .withFallback(testConfig)
+                .withFallback(userConfig)
+                .withFallback(userDirConfig)
+                .withFallback(cmdLineConfigRes)
+                .withFallback(referenceConfig);
+    }
```

Comment:
Maybe is a good time to tweak this a little so we don't create so many `ConfigFactory.empty()`. Something like
```java
Config javaSystemProperties = ConfigFactory.load("no-such-resource-only-system-props");
// ...
String res = System.getProperty("rskj.conf.res");
if (res != null) {
    javaSystemProperties = javaSystemProperties.withFallback(ConfigFactory.parseResources(res));
    logger.info("Config ( YES ): user properties from -Drskj.conf.res resource '{}'", res);
}
// ...
return javaSystemProperties;
```
but this is just a detail

Reply:
Removing of those ignored tests is fine.